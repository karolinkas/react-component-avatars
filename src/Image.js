import React, { Component } from 'react';

export class Image extends Component {
    constructor (props){
      super(props);
      this.state = {...props.image};
      this.state.selected = false;

      this.setSelected = () => {
        this.setState({selected: !this.state.selected});
      };
    }
  
    render (){
      return (<div key={this.state.src.toString()} onClick={this.props.imageToGrid} className="image-wrapper">
        <div className={"overlay-background"}></div>
        <img title="Pick me!" onClick={this.setSelected} className={["avatar", "circle-border", (this.state.selected ? "spinner" : null), (this.props.selected ? "active" : null)].join(" ")} src={this.state.src} alt="avatar"/>
        <div className={this.state.selected ? "spinner" : null}></div>
      </div>);
    }
}
