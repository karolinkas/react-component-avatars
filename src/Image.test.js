import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

import {Image} from './Image';


it('Image gets rendered with right attributes', () => {

  const image = renderer.create(<Image/>);
  expect(image).toMatchSnapshot();

});