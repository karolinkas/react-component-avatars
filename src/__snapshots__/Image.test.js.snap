// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image gets rendered with right attributes 1`] = `
<div
  className="image-wrapper"
  onClick={undefined}
>
  <div
    className="overlay-background"
  />
  <img
    alt="avatar"
    className="avatar circle-border  "
    id={undefined}
    onClick={[Function]}
    src={undefined}
    title="Pick me!"
  />
  <div
    className={null}
  />
</div>
`;
